### PR TITLE
markers linked to search results

### PR DESCRIPTION
**Pull request by cchchoi1986**
@harrychenca @paulld

### CODE REVIEW

File: www/js/controllers.js [hunk 1]
```diff
@@ -77,8 +76,36 @@ angular.module('starter.controllers', ['urlConstant'])
...
+
+  $rootScope.$on('searchResults', function (param1, param2) {
+    // $scope.trailMarkers = $scope.results;
+    // console.log($scope.trailMarkers);
+    $scope.trailMarkers = [];
+    populateMarkers($scope.trailMarkers,param2.trails)
+  });
 
+  $scope.hihi = function() {
+    console.log("hihi");
```
Comment (paulld) [inline www/js/controllers.js]: what is this?

Comment (cchchoi1986) [inline www/js/controllers.js]: will be deleted.... its just for me to test the ng-click on markers which still hasn't been done

**Comment by cchchoi1986** [comment]
also lumped in css changes including if no search results, show no result div....

linked search results to map markers

refactored $watch functions

File: www/js/controllers.js [hunk 2]
```diff
@@ -77,8 +76,36 @@ angular.module('starter.controllers', ['urlConstant'])
...
+      })
+    };
+    console.log(markers);
+  }
+
+  $http.get(apiUrl+'trails').success(function(data, status, xhr){
+    populateMarkers($scope.trailMarkers,data.trails)
+  });
+
+  $rootScope.$on('searchResults', function (param1, param2) {
```
Comment (paulld) [inline www/js/controllers.js]: you shouldn't use `param1` and `param2`.
The docs say:

```
$rootScope.$on(name, function(event, args...) { });
```

then you can use

```
$rootScope.$on('searchResults', function(event, data) { 
  // use data
});
```

File: www/js/controllers.js (hunk 1 above)
Comment (paulld) [inline www/js/controllers.js]: It belongs to another branch then?

File: www/js/controllers.js [hunk 3]
```diff
@@ -164,64 +172,23 @@ angular.module('starter.controllers', ['urlConstant'])
...
-    timeoutPromise = $timeout(function(){   //Set timeout
-      submitParams();
-      $scope.loading = false;
-    }, timeOut);
-  });
-
+  }
+  
 
+  var checkResults = function() {
```
Comment (paulld) [inline www/js/controllers.js]: Are you using this function somewhere else?
Could be written in one line:

```
$scope.noResults = $scope.results.trails.length < 1 ? true : false;
```

File: www/js/controllers.js [hunk 4]
```diff
@@ -231,12 +198,14 @@ angular.module('starter.controllers', ['urlConstant'])
   var submitParams = function() {
     $http.get(apiUrl+'search?duration='+$scope.searchParameters.duration+"&difficulty="+$scope.searchParameters.difficulty+"&scenery="+$scope.searchParameters.scenery+"&distance="+$scope.searchParameters.distance+"&hk="+$scope.searchParameters.regions[0].checked+"&kln="+$scope.searchParameters.regions[1].checked+"&nt="+$scope.searchParameters.regions[2].checked).success(function(data, status, xhr){
         $scope.results = data;
-        console.log($scope.searchParameters.regions.checked)
-        console.log($scope.results);
+        $scope.loading = false;
```
Comment (paulld) [inline www/js/controllers.js:222]: well thought

File: www/js/controllers.js [hunk 5]
```diff
@@ -231,12 +198,14 @@ angular.module('starter.controllers', ['urlConstant'])
   var submitParams = function() {
     $http.get(apiUrl+'search?duration='+$scope.searchParameters.duration+"&difficulty="+$scope.searchParameters.difficulty+"&scenery="+$scope.searchParameters.scenery+"&distance="+$scope.searchParameters.distance+"&hk="+$scope.searchParameters.regions[0].checked+"&kln="+$scope.searchParameters.regions[1].checked+"&nt="+$scope.searchParameters.regions[2].checked).success(function(data, status, xhr){
```
Comment (paulld) [inline www/js/controllers.js]: can you create an url variable above the $http request, and concatenate over several lines, for more readability?

File: www/js/controllers.js [hunk 6]
```diff
@@ -296,10 +287,19 @@ angular.module('starter.controllers', ['urlConstant'])
 
 .controller('FloraCtrl', function($scope, $http, $stateParams, $ionicModal, apiUrl) {
 
-  $http.get(apiUrl+'trails/'+$stateParams.id+'/flora').success(function(data, status, xhr){
-  // $http.get('http://localhost:3000/flora').success(function(data, status, xhr){
-      $scope.plants = data;
+  var checkPlants = function() {
```
Comment (paulld) [inline www/js/controllers.js]: You can use ternary operator

File: www/js/controllers.js (hunk 1 above)
Comment (cchchoi1986) [inline www/js/controllers.js]: yessss

File: www/js/controllers.js (hunk 3 above)
Comment (cchchoi1986) [inline www/js/controllers.js]: awesome, cool code

File: www/js/controllers.js (hunk 5 above)
Comment (cchchoi1986) [inline www/js/controllers.js]: ok done

**Pull request merged**